src/components/application_manager/rpc_plugins/rc_rpc_plugin/include/rc_rpc_plugin/rc_pending_resumption_handler.h

```diff
@@ -0,0 +1,62 @@
+#pragma once
+
+#include <list>
+#include <map>
+#include <queue>
+#include "application_manager/event_engine/event_observer.h"
+#include "application_manager/resumption/extension_pending_resumption_handler.h"
+#include "application_manager/resumption/resumption_data_processor.h"
+#include "application_manager/rpc_service.h"
+#include "rc_rpc_plugin/rc_app_extension.h"
+
+namespace rc_rpc_plugin {
+
+/**
+ * @brief The RCPendingResumptionHandler class
+ * responsibility to avoid duplication of subscription requests to HMI
+ * if multiple applications are registering
+ */
+class RCPendingResumptionHandler
+    : public resumption::ExtensionPendingResumptionHandler {
+ public:
+  RCPendingResumptionHandler(
+      application_manager::ApplicationManager& application_manager);
+
+  void on_event(const application_manager::event_engine::Event& event) override;
+
+  void HandleResumptionSubscriptionRequest(
+      application_manager::AppExtension& extension,
+      resumption::Subscriber& subscriber,
+      application_manager::Application& app) override;
+
+  void ClearPendingResumptionRequests() override;
```

Comment:
rebase to latest changes this function is renamed

Reply:
We need to update the feature/sdl_0188 branch to reflect the changes from the feature/sdl_0190, I believe this activity should be not in scope of this PR